dbms/src/Columns/tests/gtest_column_insertFrom.cpp

```diff
@@ -57,54 +57,82 @@ class TestColumnInsertFrom : public ::testing::Test
-        cols[1]->insertDisjunctFrom(*column_ptr, position_vec);
         {
-            ColumnWithTypeAndName ref(std::move(cols[0]), col_with_type_and_name.type, "");
-            ColumnWithTypeAndName result(std::move(cols[1]), col_with_type_and_name.type, "");
-            compareColumn(ref, result);
+            for (size_t i = 0; i < 2; ++i)
+            {
+                cols[i] = column_ptr->cloneEmpty();
+            }
+            std::vector<size_t> position_vec;
+            position_vec.push_back(0);
+            position_vec.push_back(2);
+            position_vec.push_back(4);
+            for (size_t position : position_vec)
+                cols[0]->insertFrom(*column_ptr, position);
+            for (size_t position : position_vec)
+                cols[0]->insertFrom(*column_ptr, position);
+            cols[1]->insertDisjunctFrom(*column_ptr, position_vec);
+            cols[1]->insertDisjunctFrom(*column_ptr, position_vec);
+            {
+                ColumnWithTypeAndName ref(std::move(cols[0]), col_with_type_and_name.type, "");
+                ColumnWithTypeAndName result(std::move(cols[1]), col_with_type_and_name.type, "");
+                compareColumn(ref, result);
+            }
         }
 
         /// Test insertManyDefaults
-        for (size_t i = 0; i < 2; ++i)
         {
-            cols[i] = column_ptr->cloneEmpty();
+            for (size_t i = 0; i < 2; ++i)
+            {
+                cols[i] = column_ptr->cloneEmpty();
+            }
+            for (size_t i = 0; i < 3; ++i)
+                cols[0]->insertDefault();
+            for (size_t i = 0; i < 3; ++i)
+                cols[0]->insertDefault();
+            cols[1]->insertManyDefaults(3);
+            cols[1]->insertManyDefaults(3);
+            {
+                ColumnWithTypeAndName ref(std::move(cols[0]), col_with_type_and_name.type, "");
+                ColumnWithTypeAndName result(std::move(cols[1]), col_with_type_and_name.type, "");
+                compareColumn(ref, result);
+            }
         }
-        for (size_t i = 0; i < 3; ++i)
-            cols[0]->insertDefault();
-        for (size_t i = 0; i < 3; ++i)
-            cols[0]->insertDefault();
-        cols[1]->insertManyDefaults(3);
-        cols[1]->insertManyDefaults(3);
+
+        /// Test insertMany
         {
-            ColumnWithTypeAndName ref(std::move(cols[0]), col_with_type_and_name.type, "");
-            ColumnWithTypeAndName result(std::move(cols[1]), col_with_type_and_name.type, "");
-            compareColumn(ref, result);
+            for (size_t i = 0; i < 2; ++i)
+                cols[i] = column_ptr->cloneEmpty();
+            for (size_t i = 0; i < 6; ++i)
+                cols[0]->insertFrom(*column_ptr, 1);
+            if unlikely (
+                typeid_cast<const ColumnNothing *>(column_ptr.get())
+                || typeid_cast<const ColumnSet *>(column_ptr.get()))
+            {
```

Comment:
```suggestion
            if (unlikely (
                typeid_cast<const ColumnNothing *>(column_ptr.get())
                || typeid_cast<const ColumnSet *>(column_ptr.get())))
            {
```

Better use `if ( unlikely(...) )`. `( ... )` is needed